### ISSUE TITLE
High severity vulnerability  in one of dependencies

### ISSUE THREAD

**Issue by pciazynski** [nodkz/mongodb-memory-server#237]
Hello! 

We are using `mongodb-memory-server` package and `npm audit` gave us information that one of the dependencies used by `mongodb-memory-server`  has high security vulnerability which requires manual intervention. 

Could you please check that? Thank you!

More info:

```
                      === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Machine-In-The-Middle                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ https-proxy-agent                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.0.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ mongodb-memory-server [dev]                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ mongodb-memory-server > mongodb-memory-server-core >         │
│               │ https-proxy-agent                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1184                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by pciazynski** [comment]
Sorry, I didn't realize that you had fixed that just 30 minutes before I opened this issue. Thx!

**Comment by nodkz** [comment]
Fixed in 5.2.11, also was fixed "maximum call stack size exceeded" for ReplicaSets.
